Fix #475 : Added an option to refer a folder of GFF files

IvanUgrin opened the pull request:
This patch adds an option to refer a folder of GFF files to the agat_sp_merge_annotations.pl script. It addresses feature request #475

Juke34 commented:
Thank you for your involvement. 
Could you update the regex to recognize gtf and gff file?
If there is a way to minimize the external library to use it would be nice. If not the new libraries have to be added in the manifest (and in the bioconda recipe. This is for remember myself)

IvanUgrin commented:
Hey, I've added GTF support and also removed the Basename import. I don't think I can remove the other two imports as they are needed to detect the files on a folder level.
